src/lib/src/service/persistence/persistence.service.spec.ts

```diff
@@ -28,21 +29,69 @@ describe('PersistenceExecutionService', () => {
 
   it('invokes REST endpoint with encoded path', fakeAsync(inject([HttpTestingController, PersistenceService],
     (httpMock: HttpTestingController, persistenceService: PersistenceService) => {
-      // given
-      let tclFilePath = 'path/to/file?.tcl';
+    // given
+    let tclFilePath = 'path/to/file?.tcl';
 
-      // when
-      persistenceService.deleteResource(tclFilePath)
+    // when
+    persistenceService.deleteResource(tclFilePath)
 
-        // then
-        .then(response => {
-          expect(response).toBe('');
-        });
+      // then
+      .subscribe(response => {
+        expect(response).toBe('');
+      });
 
-      httpMock.match({
-        method: 'DELETE',
-        url: serviceConfig.persistenceServiceUrl + '/documents/path/to/file%3F.tcl'
-      })[0].flush('');
-    })));
+    httpMock.match({
+      method: 'DELETE',
+      url: serviceConfig.persistenceServiceUrl + '/documents/path/to/file%3F.tcl'
+    })[0].flush('');
+  })));
 
+  it('createResource returns Conflict object if HTTP status code is CONFLICT',
+    inject([HttpTestingController, PersistenceService],
+    (httpMock: HttpTestingController, persistenceService: PersistenceService) => {
+    // given
+    let tclFilePath = 'path/to/file.tcl';
+    const url = `${serviceConfig.persistenceServiceUrl}/documents/${tclFilePath}`;
+    const message = `The file '${tclFilePath}' already exists.`;
+    const mockResponse = new HttpResponse({ body: message, status: 409, statusText: 'Conflict' });
+
+    const expectedResult = new Conflict(message);
+
+    // when
+    const actualObservableResult = persistenceService.createResource(tclFilePath, 'file');
+
+    // then
+    actualObservableResult.subscribe(actualResult => {
+      expect(actualResult).toEqual(expectedResult);
+    });
+
+    const actualRequest = httpMock.expectOne({ method: 'POST' });
+    expect(actualRequest.request.url).toEqual('http://localhost:9080/documents/path/to/file.tcl');
+    expect(actualRequest.request.params.get('type')).toEqual('file');
+    actualRequest.flush(message, {status: 409, statusText: 'Conflict'});
+  }));
+
+  it('deleteResource returns Conflict object if HTTP status code is CONFLICT',
+    inject([HttpTestingController, PersistenceService],
+    (httpMock: HttpTestingController, persistenceService: PersistenceService) => {
+    // given
+    let tclFilePath = 'path/to/file.tcl';
+    const url = `${serviceConfig.persistenceServiceUrl}/documents/${tclFilePath}`;
```

Comment:
is this constant used? could it be used in the 'then'-block to compare the actualRequest url?